app/controllers/suggestions.rb

```diff
@@ -0,0 +1,82 @@
+get "/suggestions" do
+  redirect "/"
+end
+
+get "/suggestions/new" do
+  current_user
+  redirect '/' unless authorized?(current_user)
```

Comment:
Heads up: the way `authorized?` is written, it is comparing the passed-in user to `current_user`. Thus, if you pass in `current_user`, it will always return true.